### PR TITLE
fix(grpc-loader): add new options by generic props loader

### PR DESCRIPTION
**Pull request by wrspada02**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Other... Please describe:

## What is the current behavior?
The current grpc loader options for custom options has type any.

Issue Number: #13930 

## What is the new behavior?
This PR allows to pass generic type to the grpc loader options, and then, the custom options provided starts having type safety.

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by wrspada02** [comment]
Hello @micalevisk!

File: packages/microservices/interfaces/client-metadata.interface.ts [hunk 1]
```diff
@@ -14,11 +14,11 @@ import {
 import { Serializer } from './serializer.interface';
 import { ConnectionOptions } from 'tls';
 
-export type ClientOptions =
+export type ClientOptions<ClientProviderCustomOptions = {}> =
```
Comment (micalevisk) [inline packages/microservices/interfaces/client-metadata.interface.ts:17]: can't we use `any` here instead of `{}`? :thinking: 

it should be less risky

Comment (wrspada02) [reply packages/microservices/interfaces/client-metadata.interface.ts]: Yes, sure. Done.

**Comment by wrspada02** [comment]
@micalevisk Is there anything else you'd like I change?

**Comment by wrspada02** [comment]
Hello @kamilmysliwiec, @micalevisk. Does this PR help your core with type safe on this grpc property?

File: packages/microservices/interfaces/microservice-configuration.interface.ts [hunk 2]
```diff
@@ -40,10 +40,24 @@ export interface CustomStrategy {
...
+  oneofs?: boolean;
+  json?: boolean;
+  includeDirs?: string[];
+}
+
 /**
  * @publicApi
  */
-export interface GrpcOptions {
+export interface GrpcOptions<GrpcLoaderOptions = any> {
```
Comment (kamilmysliwiec) [inline packages/microservices/interfaces/microservice-configuration.interface.ts:60]:
```suggestion
export interface GrpcOptions<GrpcLoaderOptions = DefaultLoaderOptions> {
```
?

Comment (wrspada02) [reply packages/microservices/interfaces/microservice-configuration.interface.ts]: Yeah, it's good too.

Then I removed ```ts GrpcLoaderOption | DefaultLoaderOption``` from line 84 to ```ts GrpcLoaderOption``` only.

Commit: https://github.com/nestjs/nest/pull/13941/commits/a4df20a31779fd4b698608536c1e112bd926fc9e

**Comment by wrspada02** [comment]
Is this missing something else? @kamilmysliwiec @micalevisk

**Pull request closed without merging**